File: frontend/src/api/recetteReamendement.ts
import client from './client'

export interface RecetteReamendementLigne {
  id_ligne?:    number
  id_produit:   number
  quantite:     number
  unite?:       string   // g | Kg | mL | L
  note_ligne?:  string
  ordre:        number
  nom_produit?: string
  type_produit?: string
}

export interface RecetteReamendement {
  id_recette_reamend: number
  nom_recette:        string
  volume_pot?:        number   // volume du pot en L
  unite_pot?:         string   // L
  notes?:             string
  lignes:             RecetteReamendementLigne[]
}

export type RecetteReamendementCreate = Omit<RecetteReamendement, 'id_recette_reamend'>
export type RecetteReamendementUpdate = Partial<RecetteReamendementCreate>

export const recetteReamendementAPI = {
  getAll:    ()                                               => client.get<RecetteReamendement[]>('/recettes-reamendement/'),
  getById:   (id: number)                                     => client.get<RecetteReamendement>(`/recettes-reamendement/${id}`),
  create:    (data: RecetteReamendementCreate)                => client.post<RecetteReamendement>('/recettes-reamendement/', data),
  update:    (id: number, data: RecetteReamendementUpdate)    => client.put<RecetteReamendement>(`/recettes-reamendement/${id}`, data),
  delete:    (id: number)                                     => client.delete(`/recettes-reamendement/${id}`),
  exportCSV: ()                                               => client.get('/recettes-reamendement/export/csv', { responseType: 'blob' }),
  importCSV: (file: File) => {
    const form = new FormData(); form.append('file', file)
    return client.post('/recettes-reamendement/import', form, { headers: { 'Content-Type': 'multipart/form-data' } })
  },
}

File: backend/app/schemas/recette_reamendement.py
"""Schemas Pydantic pour RecetteReamendement"""
from pydantic import BaseModel, ConfigDict
from typing import Optional, List


class RecetteReamendementLigneBase(BaseModel):
    id_produit: int
    quantite:   float
    unite:      Optional[str] = None   # g | Kg | mL | L
    note_ligne: Optional[str] = None
    ordre:      int = 0


class RecetteReamendementLigneCreate(RecetteReamendementLigneBase):
    pass


class RecetteReamendementLigneRead(RecetteReamendementLigneBase):
    id_ligne:     int
    nom_produit:  Optional[str] = None
    type_produit: Optional[str] = None
    model_config  = ConfigDict(from_attributes=True)


class RecetteReamendementBase(BaseModel):
    nom_recette: str
    volume_pot:  Optional[float]= None   # volume du pot en L
    unite_pot:   Optional[str]  = "L"
    notes:       Optional[str]  = None


class RecetteReamendementCreate(RecetteReamendementBase):
    lignes: List[RecetteReamendementLigneCreate] = []


class RecetteReamendementUpdate(BaseModel):
    nom_recette: Optional[str]  = None
    volume_pot:  Optional[float]= None
    unite_pot:   Optional[str]  = None
    notes:       Optional[str]  = None
    lignes:      Optional[List[RecetteReamendementLigneCreate]] = None


class RecetteReamendementRead(RecetteReamendementBase):
    id_recette_reamend: int
    lignes:             List[RecetteReamendementLigneRead] = []
    model_config        = ConfigDict(from_attributes=True)
